src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -92,157 +75,184 @@ internal sealed class ParsedStackFrame : ParsedFrame
-        /// ex: at ConsoleApp4.MyClass.M[T](T t)[| in |]C:\repos\Test\MyClass.cs:line 7
-        /// </summary>
-        public string? GetTextBetweenTypeAndFile()
-        {
-            if (FileSpan == default)
+            for (var i = 0; i < stackFrameParameters.Parameters.Length; i++)
             {
-                return null;
+                var stackFrameParameter = stackFrameParameters.Parameters[i];
+                var paramSymbol = parameters[i];
+
+                if (paramSymbol.Name != stackFrameParameter.Identifier.ToString(skipTrivia: true))
+                {
+                    return false;
+                }
+
+                if (!MatchType(paramSymbol.Type, stackFrameParameter.Type))
+                {
+                    return false;
+                }
             }
 
-            return OriginalText[(ArgsSpan.End + 1)..FileSpan.Start];
+            return true;
         }
 
-        internal (Document? document, int line) GetDocumentAndLine(Solution solution)
+        private static bool MatchTypeArguments(ImmutableArray<ITypeSymbol> typeArguments, StackFrameTypeArgumentList? stackFrameTypeArgumentList)
         {
-            var fileMatches = GetFileMatches(solution, out var lineNumber);
-            if (fileMatches.IsEmpty)
+            if (stackFrameTypeArgumentList is null)
             {
-                return (null, 0);
+                return typeArguments.IsDefaultOrEmpty;
             }
 
-            return (fileMatches.First(), lineNumber);
+            if (typeArguments.IsDefaultOrEmpty)
+            {
+                return false;
+            }
+
+            var stackFrameTypeArguments = stackFrameTypeArgumentList.TypeArguments;
+            return typeArguments.Length == stackFrameTypeArguments.Length;
         }
 
-        /// <summary>
-        /// If the <see cref="FileSpan"/> exists it attempts to map the file path that exists
-        /// in that span to a document in the solution. It's possible the file path won't match exactly,
-        /// so it does a best effort to match based on name.
-        /// </summary>
-        private ImmutableArray<Document> GetFileMatches(Solution solution, out int lineNumber)
+        private static bool MatchType(ITypeSymbol type, StackFrameTypeNode stackFrameType)
         {
-            var fileText = OriginalText[FileSpan.Start..FileSpan.End];
-            var regex = new Regex(@"(?<fileName>.+):(line)\s*(?<lineNumber>[0-9]+)");
-            var match = regex.Match(fileText);
-            Debug.Assert(match.Success);
-
-            var fileNameGroup = match.Groups["fileName"];
-            var lineNumberGroup = match.Groups["lineNumber"];
-
-            lineNumber = int.Parse(lineNumberGroup.Value);
-
-            var fileName = fileNameGroup.Value;
-            Debug.Assert(!string.IsNullOrEmpty(fileName));
+            if (type is IArrayTypeSymbol arrayType)
```

Comment:
nit: i have a feeling this outer if is unnecessary, and we can just have the foreach/while loop inside that unraps the syntax while validating it against the type symbol.

Reply:
Not sure I understand this comment. Could you elaborate or just provide pseudocode?

Reply:
taking offline.